llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -4025,17 +4028,33 @@ OpenMPIRBuilder::InsertPointTy OpenMPIRBuilder::createSingle(
   Function *ExitRTLFn = getOrCreateRuntimeFunctionPtr(OMPRTL___kmpc_end_single);
   Instruction *ExitCall = Builder.CreateCall(ExitRTLFn, Args);
 
+  auto FiniCBWrapper = [&](InsertPointTy IP) {
+    FiniCB(IP);
+
+    if (DidIt)
+      Builder.CreateStore(Builder.getInt32(1), DidIt);
```

Comment:
Nit: A comment here also as to why this is being set to one here. Because only the single thread executes the inlined single region.